src/codeu/chat/util/ServerLog.java

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.chat.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import codeu.chat.server.Controller;
+
+public final class ServerLog {
+  private File log;	
+  private HashMap<Integer,String> lines;
```

Comment:
Use `Map` instead of `HashMap` here. Generally, it's better to use interface classes for variables. Example:

`List<Integer> = new ArrayList<>();`